<?xml version="1.0" ?>
<sdf version="1.6">
  <world name="default">
    
    <!-- Physics settings for better simulation -->
    <physics type="ode">
      <max_step_size>0.001</max_step_size>
      <real_time_factor>1</real_time_factor>
      <real_time_update_rate>1000</real_time_update_rate>
    </physics>
    
    <!-- Scene settings for professional visuals -->
    <scene>
      <ambient>0.6 0.6 0.6 1</ambient>
      <background>0.15 0.15 0.2 1</background>
      <shadows>true</shadows>
      <fog>
        <color>0.15 0.15 0.2 1</color>
        <type>linear</type>
        <start>10</start>
        <end>30</end>
        <density>0.1</density>
      </fog>
      <sky>
        <clouds>
          <speed>12</speed>
        </clouds>
      </sky>
    </scene>
    
    <!-- Professional studio lighting setup -->
    <light name='main_sun' type='directional'>
      <cast_shadows>1</cast_shadows>
      <pose>0 0 10 0 0 0</pose>
      <diffuse>0.9 0.9 0.9 1</diffuse>
      <specular>0.4 0.4 0.4 1</specular>
      <attenuation>
        <range>1000</range>
        <constant>0.9</constant>
        <linear>0.01</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
      <direction>-0.3 0.3 -0.9</direction>
    </light>
    
    <light name='key_light' type='point'>
      <cast_shadows>0</cast_shadows>
      <pose>2 2 3 0 0 0</pose>
      <diffuse>0.8 0.8 1 1</diffuse>
      <specular>0.8 0.8 0.8 1</specular>
      <attenuation>
        <range>20</range>
        <constant>0.5</constant>
        <linear>0.01</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
    </light>
    
    <light name='fill_light' type='point'>
      <cast_shadows>0</cast_shadows>
      <pose>-2 -2 2 0 0 0</pose>
      <diffuse>0.4 0.5 0.6 1</diffuse>
      <specular>0.2 0.2 0.2 1</specular>
      <attenuation>
        <range>20</range>
        <constant>0.3</constant>
        <linear>0.01</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
    </light>
    
    <include>
      <uri>model://sun</uri>
    </include>
    
    <!-- Modern textured ground plane -->
    <model name='ground_plane'>
      <static>true</static>
      <link name='link'>
        <collision name='collision'>
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>100 100</size>
            </plane>
          </geometry>
          <surface>
            <friction>
              <ode>
                <mu>100</mu>
                <mu2>50</mu2>
              </ode>
            </friction>
          </surface>
        </collision>
        <visual name='visual'>
          <cast_shadows>0</cast_shadows>
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>100 100</size>
            </plane>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/DarkGrey</name>
            </script>
          </material>
        </visual>
      </link>
    </model>

    <!-- Professional wooden table with realistic appearance -->
    <model name="table">
      <static>true</static>
      <link name="table_link">
        <pose>0.5 0 0.025 0 0 0</pose>
        <visual name="table_visual">
          <cast_shadows>1</cast_shadows>
          <geometry>
            <box>
              <size>0.8 1.2 0.05</size>
            </box>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/Wood</name>
            </script>
          </material>
        </visual>
        <collision name="table_collision">
          <geometry>
            <box>
              <size>0.8 1.2 0.05</size>
            </box>
          </geometry>
        </collision>
      </link>
      <!-- Cylindrical wooden table legs with texture -->
      <link name="leg1">
        <pose>0.35 0.55 -0.35 0 0 0</pose>
        <visual name="leg1_visual">
          <cast_shadows>1</cast_shadows>
          <geometry>
            <cylinder>
              <radius>0.03</radius>
              <length>0.7</length>
            </cylinder>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/Wood</name>
            </script>
          </material>
        </visual>
      </link>
      <link name="leg2">
        <pose>0.65 0.55 -0.35 0 0 0</pose>
        <visual name="leg2_visual">
          <cast_shadows>1</cast_shadows>
          <geometry>
            <cylinder>
              <radius>0.03</radius>
              <length>0.7</length>
            </cylinder>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/Wood</name>
            </script>
          </material>
        </visual>
      </link>
      <link name="leg3">
        <pose>0.35 -0.55 -0.35 0 0 0</pose>
        <visual name="leg3_visual">
          <cast_shadows>1</cast_shadows>
          <geometry>
            <cylinder>
              <radius>0.03</radius>
              <length>0.7</length>
            </cylinder>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/Wood</name>
            </script>
          </material>
        </visual>
      </link>
      <link name="leg4">
        <pose>0.65 -0.55 -0.35 0 0 0</pose>
        <visual name="leg4_visual">
          <cast_shadows>1</cast_shadows>
          <geometry>
            <cylinder>
              <radius>0.03</radius>
              <length>0.7</length>
            </cylinder>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/Wood</name>
            </script>
          </material>
        </visual>
      </link>
    </model>

    <!-- Spawn the robot model -->
    <include>
      <uri>model://simple_3dof_arm</uri>
      <pose>0 0 0 0 0 0</pose>
    </include>

    <!-- Attractive target cube with glossy red finish -->
    <model name="cube_object">
      <static>false</static>
      <link name="box_link">
        <pose>0.5 0.15 0.075 0 0 0</pose>
        <inertial>
          <mass>0.15</mass>
          <inertia>
            <ixx>0.000375</ixx>
            <iyy>0.000375</iyy>
            <izz>0.000375</izz>
          </inertia>
        </inertial>
        <visual name="box_visual">
          <cast_shadows>1</cast_shadows>
          <geometry>
            <box>
              <size>0.05 0.05 0.05</size>
            </box>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/Red</name>
            </script>
          </material>
        </visual>
        <collision name="box_collision">
          <geometry>
            <box>
              <size>0.05 0.05 0.05</size>
            </box>
          </geometry>
          <surface>
            <friction>
              <ode>
                <mu>1.0</mu>
                <mu2>1.0</mu2>
              </ode>
            </friction>
            <bounce>
              <restitution_coefficient>0.1</restitution_coefficient>
            </bounce>
          </surface>
        </collision>
      </link>
    </model>
    
    <!-- Add some extra objects for visual interest -->
    <model name="cylinder_object">
      <static>false</static>
      <pose>0.5 -0.2 0.075 0 0 0</pose>
      <link name="cyl_link">
        <inertial>
          <mass>0.2</mass>
        </inertial>
        <visual name="cyl_visual">
          <cast_shadows>1</cast_shadows>
          <geometry>
            <cylinder>
              <radius>0.03</radius>
              <length>0.1</length>
            </cylinder>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/Blue</name>
            </script>
          </material>
        </visual>
        <collision name="cyl_collision">
          <geometry>
            <cylinder>
              <radius>0.03</radius>
              <length>0.1</length>
            </cylinder>
          </geometry>
        </collision>
      </link>
    </model>
    
    <model name="sphere_object">
      <static>false</static>
      <pose>0.65 0 0.065 0 0 0</pose>
      <link name="sphere_link">
        <inertial>
          <mass>0.1</mass>
        </inertial>
        <visual name="sphere_visual">
          <cast_shadows>1</cast_shadows>
          <geometry>
            <sphere>
              <radius>0.03</radius>
            </sphere>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/Green</name>
            </script>
          </material>
        </visual>
        <collision name="sphere_collision">
          <geometry>
            <sphere>
              <radius>0.03</radius>
            </sphere>
          </geometry>
        </collision>
      </link>
    </model>

  </world>
</sdf>
